relay-server/src/services/processor/playstation.rs

```diff
@@ -2,42 +2,333 @@
+            Annotated::new(app_info.version.to_owned());
     }
 
-    Ok(())
+    contexts.add(device_context);
+    contexts.add(os_context);
+    contexts.add(runtime_context);
 }
 
-pub fn process(
-    _managed_envelope: &mut TypedEnvelope<ErrorGroup>,
-    _event: &mut Annotated<Event>,
-) -> Result<Option<EventFullyNormalized>, ProcessingError> {
-    // TODO: Add the processing logic here.
-    Ok(None)
+fn infer_content_type(filename: &str) -> ContentType {
+    // Since we only receive a limited selection of files through this mechanism this simple logic
+    // should be enough.
+    let extension = filename.rsplit('.').next().map(str::to_lowercase);
+    match extension.as_deref() {
+        Some("txt") => ContentType::Text,
+        Some("json") => ContentType::Json,
+        Some("xml") => ContentType::Xml,
+        _ => ContentType::OctetStream,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::borrow::Cow;
+    use std::collections::BTreeMap;
+
+    use relay_protocol::Value;
+
+    use insta::assert_debug_snapshot;
+
+    #[test]
+    fn test_event_release_setting() {
+        let prospero = ProsperoDump {
+            userdata: BTreeMap::from([("release", Cow::Borrowed("1.2.3.4"))]),
+            ..Default::default()
+        };
+
+        let mut event = Event::default();
+        update_sentry_event(&mut event, &prospero);
+        assert_eq!(
+            event.release,
+            Annotated::new(LenientString("1.2.3.4".to_owned()))
+        );
+    }
+
+    #[test]
+    fn test_event_field_mapping() {
+        let prospero = ProsperoDump {
+            userdata: BTreeMap::from([
+                ("sentry.release", Cow::Borrowed("1.2.3.4")),
+                ("sentry.environment", Cow::Borrowed("production")),
+                ("sentry.user.username", Cow::Borrowed("janedoe")),
+                ("sentry.user.email", Cow::Borrowed("janedoe@example.com")),
+                ("santry.tag", Cow::Borrowed("other_value")),
+                ("other_tag", Cow::Borrowed("other_value")),
+            ]),
+            ..Default::default()
+        };
+
+        let mut event = Event::default();
+        update_sentry_event(&mut event, &prospero);
+
+        // Check the event fields
+        assert_eq!(
+            event.release,
+            Annotated::new(LenientString("1.2.3.4".to_owned()))
+        );
+        assert_eq!(event.environment, Annotated::new("production".to_owned()));
+
+        if let Some(user) = event.user.0 {
+            assert_eq!(
+                user.username,
+                Annotated::new(LenientString("janedoe".to_owned()))
+            );
+            assert_eq!(user.email, Annotated::new("janedoe@example.com".to_owned()));
+        } else {
+            panic!("User information not set in the event");
+        }
+
+        // Checks the tags are correct
+        let tags = event.tags.value().unwrap();
+        assert_debug_snapshot!(tags);
```

Comment:
I would assert this inline, `assert_debug_snapshot!(tags, @r"")`, makes following the test easier and the output isn't that large.

Or assert a json snapshot, that makes it a bit more concise.